frontends/p4/sideEffects.cpp

```diff
@@ -51,9 +51,10 @@ struct EvaluationOrder {
     }
 
     const IR::Expression* addAssignment(
+        Util::SourceInfo srcInfo,
         cstring varName, const IR::Expression* expression) {
         auto left = new IR::PathExpression(IR::ID(varName, nullptr));
-        auto stat = new IR::AssignmentStatement(left, expression);
+        auto stat = new IR::AssignmentStatement(srcInfo, left, expression);
```

Comment:
The error message was not displaying the source fragment.
These changes preserve source position for more constructs, but they are not enough - there are other places in the code where source position information is lost.